scipy/stats/tests/test_multicomp.py

```diff
@@ -220,43 +252,106 @@ def test_ttest_ind(self, alternative):
             assert_allclose(res.pvalue, ref.pvalue, atol=1e-3)
 
     @pytest.mark.parametrize(
-        'alternative, statistic, pvalue',
+        'alternative, pvalue',
         [
-            ('less', 'low', 0),
-            ('less', 'high', 1),
-            ('greater', 'low', 1),
-            ('greater', 'high', 0),
-            ('two-sided', 'low', 0),
-            ('two-sided', 'high', 0)
+            ('less', [0, 1]),
+            ('greater', [1, 0]),
+            ('two-sided', [0, 0]),
         ]
     )
-    def test_alternatives(self, alternative, statistic, pvalue):
+    def test_alternatives(self, alternative, pvalue):
         rng = np.random.default_rng(114184017807316971636137493526995620351)
 
         for _ in range(10):
-            sample_1 = rng.integers(0, 20, size=(10,))
-            sample_2 = rng.integers(80, 100, size=(10,))
-
-            if statistic == 'high':
-                sample = sample_2
-                control = sample_1
-            else:
-                sample = sample_1
-                control = sample_2
+            # width of 20 and min diff between samples/control is 60
+            # and maximal diff would be 100
+            sample_less = rng.integers(0, 20, size=(10,))
+            control = rng.integers(80, 100, size=(10,))
+            sample_greater = rng.integers(160, 180, size=(10,))
 
             res = stats.dunnett(
-                sample, control=control,
+                sample_less, sample_greater, control=control,
                 alternative=alternative, random_state=rng
             )
             assert_allclose(res.pvalue, pvalue, atol=1e-7)
 
+            ci_ = res.confidence_interval()
+            # two-sided is comparable for high/low
+            ci_ = ci_.high if alternative == 'less' else ci_.low
```

Comment:
I'm going to revise this a bit. I understand what is going on now and that you were keeping it compact, but it surprised me that only the lower side of the two-sided confidence interval is tested. We should also be checking that the infinities are where we expect them to be.